### PR TITLE
feat/GWW-139: Create zip with csv and geojson for selected area

### PR DESCRIPTION
**Pull request by lsarni**
Ticket: [GWW-139](https://issuetracker.deltares.nl/browse/GWW-139)

**Main features**:
- Download geojson with `FeatureCollection` for basins/adm region/custom selections with multiple reservoirs
- Download the csv of the sum of the surface water area for a basin/adm region/custom selection with more than 10 reservoirs
- Download zip with the cvs of the sum, and a csv per reservoir for a basin/adm region/custom selection with less than 10 reservoirs
- Download csv/geojson file for a single reservoir page
- Hide the download options when there are no reservoirs in an area

### CODE REVIEW

File: src/components/PageExport/PageExport.vue [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+          const { name, content } = this.timeSeriesCsv(this.timeSeries.series[0])
+
+          this.downloadFile(
+            `${name}.csv`,
+            new Blob([content], { type: 'text/csv;charset=utf-8' }),
+          )
+        } else {
+          const zip = this.timeSeriesZip()
+
+          const content = await zip.generateAsync({ type: 'blob' })
```
Comment (SiggyF) [inline src/components/PageExport/PageExport.vue:105]: nice job here.

File: src/components/PageExport/PageExport.vue [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            content,
+          )
+        }
+      },
+      exportGeometry () {
+        if (this.reservoirs.length === 1) {
+          const geometry = JSON.stringify(this.reservoirs[0])
+
+          this.downloadFile(
+            this.reservoirs[0]?.properties?.name
```
Comment (SiggyF) [inline src/components/PageExport/PageExport.vue:117]: this might need a comment

File: src/components/PageExport/PageExport.vue [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          return this.timeSeriesCsv(serie)
+        })
+        files.forEach(({ name, content }) => {
+          zip.file(`${name}.csv`, content)
+        })
+        return zip
+      },
+      downloadFile (filename, blob) {
+        const anchor = document.createElement('a')
+        anchor.href = (window.URL || window.webkitURL).createObjectURL(blob)
```
Comment (SiggyF) [inline src/components/PageExport/PageExport.vue:154]: why is webkitURL needed here?

**Review by SiggyF — APPROVED** [review]
Looks good.

File: src/components/PageExport/PageExport.vue (hunk 2 above)
Comment (lsarni) [reply src/components/PageExport/PageExport.vue:117]: I have moved it to a separate method and added a comment there

File: src/components/PageExport/PageExport.vue (hunk 3 above)
Comment (lsarni) [reply src/components/PageExport/PageExport.vue:154]: Firefox uses `window.URL` and Chrome `window.webkitURL`, it's common practice to write it like this

**Pull request merged**